test/integration/targets/git/tasks/separate-git-dir.yml

```diff
@@ -0,0 +1,18 @@
+# test code for repositories with separate git dir updating
+# see https://github.com/ansible/ansible/pull/38016
+# see https://github.com/ansible/ansible/issues/30034
+
+- name: SEPARATE-GIT-DIR | clear checkout_dir
+  file:
+    state: absent
+    path: '{{ checkout_dir }}'
+
+- name: SEPARATE-GIT-DIR | clone with a separate git dir
+  command: git clone {{ repo_format1 }} git --separate-git-dir=separate.git
```

Comment:
For safety reasons, I'd create some random dir using `mktemp -d`

Reply:
A random dir for `separate.git` directory? What kind of safety do you mean? If the problem is that it may already exist then we can clear the directory the same way as the `checkout_dir`.

Reply:
yes, i'm concerned about it already existing in place, but also current approach will cause conflicts and false failures in case of simultaneous test runs.

Reply:
Why not use `{{ checkout_dir }}` instead of `git` arg + chdir?

Reply:
After adding support for module arg, you'll need to just use the git module itself rather then shell command.